eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala

```diff
@@ -24,13 +28,49 @@ case class GetInfoResult(
 case class PeerInfo(
     nodeId: NodeId,
     state: PeerState,
-    //address: String,
+    address: Option[String],
     channels: Int)
 
-case class ChannelInfo(
+/**
+  * This is the data model returned by the RPC call
+  * `channels nodeId`. The content of the objects
+  * being returne differ based on whatever state
+  * the channel is in. The member of this abstract
+  * class are in eveyr channel state, whereas other
+  * channel states may have extra information.
+  */
+sealed abstract class ChannelInfo {
+  def nodeId: NodeId
+  def channelId: FundedChannelId
+  def localMsat: MilliSatoshis
+  def remoteMsat: MilliSatoshis
+  def state: ChannelState
+}
+
+/**
+  * This represents the case where the channel is
```

Comment:
```
/**
This represents the case where the [[ChannelState]] 
is undetermined
*/
```
right?

Reply:
Unfortunately we have to do `[[org.bitcoins.core.protocol.ln.channel.ChannelState ChannelState]]`...
Adding it